tree/tree/src/TLeaf.cxx

```diff
@@ -151,6 +151,40 @@ void TLeaf::FillBasket(TBuffer &)
 }
 
 ////////////////////////////////////////////////////////////////////////////////
+/// If the class supports it, generate an offset array base.
+///
+/// This class only returns `nullptr` on error.
+Int_t *TLeaf::GenerateOffsetArrayBase(Int_t base, Int_t events) const
+{
+   // In order to avoid a virtual call, we assume ROOT developers will override
+   // the default GenerateOffsetArray for cases where this function does not apply.
+
+   Int_t *retval = new Int_t[events];
+   if (R__unlikely(!retval)) {
+      return nullptr;
+   }
+
+   if (fLeafCount) {
+      Long64_t orig_entry = std::max(fBranch->GetReadEntry(), 0LL); // -1 indicates to start at the beginning?
+      Long64_t orig_leaf_entry = fLeafCount->GetBranch()->GetReadEntry();
+      Int_t len = 0;
+      for (Int_t idx = 0, offset = base; idx < events; idx++) {
+         retval[idx] = offset;
+         fLeafCount->GetBranch()->GetEntry(orig_entry + idx);
+         len = static_cast<Int_t>(fLeafCount->GetValue());
+         offset += fLenType * len;
+      }
+      fLeafCount->GetBranch()->GetEntry(orig_leaf_entry);
+   } else {
+      for (Int_t idx = 0, offset = base; idx < events; idx++, offset += (fLenType * fLen)) {
```

Comment:
When the size is fixed, the TBasket should not need any offset array.

Reply:
Correct.  That code is now dead, given that this is the current implementation of `CanGenerateOffsetArray`:

```
virtual Bool_t CanGenerateOffsetArray() {return fLeafCount;}
```

Will remove it momentarily.